### PR TITLE
[GOBBLIN-1597] Add error handling in dagmanager to continue if dag fails to process,…

### PR DESCRIPTION
**Pull request by Will-Lo**
… make Azkaban client retry on timeouts

Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1597

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
If there is any error with the DagManager processing, it will abort all jobs processed in the thread and restart from the beginning, which could leave jobs in a permanently stuck state

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java [hunk 1]
```diff
@@ -709,48 +709,55 @@ private void pollAndAdvanceDag() throws IOException, ExecutionException, Interru
...
+          if (jobStatus != null && jobStatus.isShouldRetry()) {
+            log.info("Retrying job: {}, current attempts: {}, max attempts: {}", DagManagerUtils.getFullyQualifiedJobName(node),
+                jobStatus.getCurrentAttempts(), jobStatus.getMaxAttempts());
+            submitJob(node);
+          }
+        } catch (Exception e) {
+            // Error occurred while processing dag, leave dag for next thread execution to process
+            log.error(String.format("Exception caught in DagManager while processing dag for flowGroup %s flowName %s due to ",
+                node.getValue().getJobSpec().getConfig().getString(ConfigurationKeys.FLOW_GROUP_KEY),
+                node.getValue().getJobSpec().getConfig().getString(ConfigurationKeys.FLOW_NAME_KEY)), e);
```
Comment (phet) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java:759]: maybe `DagManagerUtils.getFullyQualifiedJobName` or `.getFlowId` (If you don't also want the exec id in there)?

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java [hunk 2]
```diff
@@ -709,48 +709,55 @@ private void pollAndAdvanceDag() throws IOException, ExecutionException, Interru
...
-              jobStatus.getCurrentAttempts(), jobStatus.getMaxAttempts());
-          submitJob(node);
-        }
+          if (jobStatus != null && jobStatus.isShouldRetry()) {
+            log.info("Retrying job: {}, current attempts: {}, max attempts: {}", DagManagerUtils.getFullyQualifiedJobName(node),
+                jobStatus.getCurrentAttempts(), jobStatus.getMaxAttempts());
+            submitJob(node);
+          }
+        } catch (Exception e) {
+            // Error occurred while processing dag, leave dag for next thread execution to process
```
Comment (phet) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java:756]: this is true... but the bigger difference, after this change, is to "continue processing other dags assigned to this thread"
(not essential comment to leave, but might help maintainers who are reading quickly)

**Review by phet — APPROVED** [review]
good change.  couple minor suggestions.

...and while a test may not seem essential, just checking whether there's a quick way to put one together... let us know!

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java (hunk 1 above)
Comment (Will-Lo) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java]: Yeah I thought about omitting the flow execution ID but if we are searching logs by flowName it should still be caught anyways so this is better cause we can also search by exec ID

**Review by phet — APPROVED** [review]

**Review by ZihanLi58 — APPROVED** [review]
+1

**Pull request merged**